File: FE/src/app/models/book.ts
import { Author } from "../components/author/author";
import { Publisher } from "../components/publisher/publisher";

export class Book {
    bookId: string = '';
    isbn: string = '';
    title: string = '';
    description: string = '';
    genre: string = '';
    coverUrl: string = '';

    publisherId: string = '';
    authorsIds: string[] = [];
    bookCopies: BookCopy[] = [];
}

export class BookViewModel {
    bookId: string = '';
    isbn: string = '';
    title: string = '';
    description: string = '';
    genre: string = '';
    coverUrl: string = '';
    blobId:string = '';

    publisher: Publisher = new Publisher();
    authors: Author[] = [];
    bookCopies: BookCopy[] = [];
}

export class BookCopy {
    bookItemId: string = '';
    bookId: string = '';
    barcode: string = '';
    bookStatus: BookCopyStatus = 0;

    borrowedDate!: Date;
    returnDate!: Date;
}

export class Genre {
    name: string = '';
}

export enum BookCopyStatus { 
    Available,
    Reserved,
    Borrowed,
    NotAvailable
}

File: BE/BusinessLayer/Models/Book.cs
using System.ComponentModel.DataAnnotations;

namespace BusinessLayer.Models;

public class Book
{
    public Guid BookId { get; set; }
    
    public string Isbn { get; set; } = string.Empty;

    public string Title { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public string Genre { get; set; } = string.Empty;

    public string CoverUrl { get; set; } = string.Empty;

    public Guid? PublisherId { get; set; }
    public Guid? BlobId { get; set; }

    public ICollection<Guid>? AuthorsIds { get; set; }

    public ICollection<BookItem>? BookItems { get; set; }
    public DateTime CreatedAt { get; set; }
}